File: leetcode/1431.kids-with-the-greatest-number-of-candies.py
#
# @lc app=leetcode id=1431 lang=python3
#
# [1431] Kids With the Greatest Number of Candies
#

# @lc code=start
class Solution:
    def kidsWithCandies(self, candies: list[int], extraCandies: int) -> list[bool]:
        limit = max(candies) - extraCandies
        return [c >= limit for c in candies]
        
# @lc code=end

File: leetcode/1431.kids-with-the-greatest-number-of-candies.kt
/*
 * @lc app=leetcode id=1431 lang=kotlin
 *
 * [1431] Kids With the Greatest Number of Candies
 */

// @lc code=start
class Solution {
    fun kidsWithCandies(candies: IntArray, extraCandies: Int): List<Boolean> {
       var maximum = 0
       for (c in candies) {
         if (c > maximum) {
           maximum = c
         }
       }
       val limit = maximum - extraCandies
       val result = mutableListOf<Boolean>()
       for (c in candies) {
         result.add(c >= limit)
       }
       return result 
        
    }
}
// @lc code=end

